src/tribler/core/utilities/path_util.py

```diff
@@ -36,8 +38,23 @@ def normalize_to(self, base: str = None) -> Path:
 
         return self
 
-    def size(self) -> int:
-        return self.stat().st_size
+    def size(self, only_files: bool = False) -> int:
+        """ Return the size of this file or directory (recursively).
+
+        Args:
+            only_files: If True, only return the size of files in this directory, not the size of files+directories.
+
+        Returns: The size of this file or directory.
+        """
+        if self.is_file():
+            return self.stat().st_size
+
+        size = 0
+        for root, dir_names, file_names in os.walk(self):
+            names = file_names if only_files else itertools.chain(dir_names, file_names)
+            paths = (os.path.join(root, name) for name in names)
+            size += sum(os.path.getsize(p) for p in paths if os.path.exists(p))
```

Comment:
A very small nitpick, the function does not take into account the size of the top-level root directory entry. To be absolutely correct, it should be something like:
```suggestion
        for root, dir_names, file_names in os.walk(self):
            if not only_files and os.path.exists(root):
                size += os.path.getsize(root)  # add the size of the directory entry
            for file_name in file_names:
                file_path = os.path.join(root, file_name)
                if os.path.exists(file_path):
                    size += os.path.getsize(file_path)
```

Reply:
What about:

```python
    def size(self, only_files: bool = False) -> int:
        """ Return the size of this file or directory (recursively).

        Args:
            only_files: If True, only return the size of files in this directory, not the size of files+directories.

        Returns: The size of this file or directory.
        """
        if not self.exists():
            return 0
        
        if self.is_file():
            return self.stat().st_size

        size = os.path.getsize(self.absolute())
        for root, dir_names, file_names in os.walk(self):
            names = file_names if only_files else itertools.chain(dir_names, file_names)
            paths = (os.path.join(root, name) for name in names)
            size += sum(os.path.getsize(p) for p in paths if os.path.exists(p))
        return size
```

Reply:
Done